### PR TITLE
Support App Store Connect API keys and provisioning profiles

### LINKED ISSUE #71: Support App Store Connect API Keys with Xcode >= 13
From the [Xcode 13 Release Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-13-release-notes):
> `xcodebuild` now supports the use of App Store Connect API keys for authentication with the Apple Developer website. This enables the use of automatic signing via `xcodebuild` in headless environments, such as build machines and continuous integration setups. To use API keys with `xcodebuild`, create an API key on [App Store Connect](https://appstoreconnect.apple.com/access/api) and pass the key along with its identifier and your team’s issuer identifier to `xcodebuild` using the new parameters `authenticationKeyPath`,` authenticationKeyID`, and `authenticationKeyIssuerID`, respectively. When creating a key, you can assign it a role to control its permissions for performing automatic signing tasks. To learn more about creating and managing keys, see [Creating API Keys for App Store Connect API](https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api). (51444716)

This looks like it might allow for a better way to enable automatic code signing when invoking Xcode 13 and later.

### PR DESCRIPTION
**Pull request by chris-araman**
Closes #71.

Using an App Store Connect API key with Xcode >= 13 allows automatic device registration, certificate creation, and provisioning profile creation.

Users who want to use an existing provisioning profile, or who use Xcode < 13, can now pass a list of Base64-encoded Mac or iOS provisioning profiles.

### CODE REVIEW

**Comment by chris-araman** [comment]
I might be able to address the provisioning profile(s) with [this](https://docs.github.com/en/actions/deployment/deploying-xcode-applications/installing-an-apple-certificate-on-macos-runners-for-xcode-development#creating-secrets-for-your-certificate-and-provisioning-profile), assuming Xcode will search for local profiles before checking the API for registration.

**Comment by chris-araman** [comment]
```
❌  error: Revoke certificate: Your account already has an Apple Development signing certificate for this machine, but its private key is not installed in your keychain. Xcode can create a new one after revoking your existing certificate
```

Looks like I may need to make use of [actions/cache](https://github.com/actions/cache).

**Comment by mxcl** [comment]
We should bump to `v2` since we are removing `code-sign-certificate`.

Unless we can reasonably prove this only effects <2 projects. (Not sure we can prove anything).

**Comment by chris-araman** [comment]
I'm not planning to remove any inputs. I reorganized the `README` section on code signing a bit:
- *App Store Connect API key*: Requires Xcode 13. Manages device registration, certificates and provisioning profiles automatically.
- *Certificates*: Used to install a specific certificate to the macOS Keychain.
- *Provisioning Profiles*: Used to install specific provisioning profiles.

None of these should be mutually exclusive, in theory. The simplest usage for most users should be the API key alone.

To Do:

- [ ] Figure out how to address the `private key is not installed` error.
  - We could find a way to persist the certificate for a given GitHub Actions host (macOS Device UUID), perhaps using [@actions/cache](https://github.com/actions/cache). What are the security implications of that?
  - Alternatively, we could call the App Store Connect API directly to revoke the existing certificate beforehand. This might have the effect of generating a long list of revoked certs on the developer's account.
  - Failing those ideas, maybe this is only useful for GHES and self-hosted runners? That would be disappointing.
- [ ] Determine whether installing a Mac `.provisionprofile` is useful, or if only iOS-ish `.mobileprovision` profiles are useful. It's not clear to me, for instance, if a provisioning profile from a developer's Mac is meant to (or can) be copied to another Mac and remain useful. Mobile profiles clearly seem useful, as there are dozens of projects copying them around, and GitHub [documents](https://docs.github.com/en/actions/deployment/deploying-xcode-applications/installing-an-apple-certificate-on-macos-runners-for-xcode-development#creating-secrets-for-your-certificate-and-provisioning-profile) how to install them. Perhaps a `.provisionprofile` is only useful for Mac Catalyst development? I that's the case, I'll document it as such.

File: src/index.ts [hunk 1]
```diff
@@ -103,6 +110,44 @@ async function main() {
     }
   }
 
+  async function configureApiKey(): Promise<string[] | false> {
```
Comment (mxcl) [inline src/index.ts:113]: Can we name this more specifically? I have read this line several times and I have to look back to the description of the PR to remember what API it is for.

**Review by mxcl — APPROVED** [review]
Safely additive.

File: src/index.ts (hunk 1 above)
Comment (chris-araman) [reply src/index.ts:113]: Sure thing.

Comment (chris-araman) [reply src/index.ts]: I renamed it to `getApiKey` to mimic the existing accessors. My concern with this is that it has the side effect of `createApiKeyFile`, but it now more clearly describes its return value. I think this is probably fine.

Comment (mxcl) [reply src/index.ts]: Well I mean, *what* API? `getAppStoreConnectApiKey` is what I was going for.

Comment (chris-araman) [reply src/index.ts]: Ah, gotcha. Addressed in latest push.

**Comment by chris-araman** [comment]
Sorry for the delay. It's been, uh, an unexpectedly busy few weeks. 😅

Still investigating the To Dos above.

**Comment by mxcl** [comment]
nps dude; to me you’re an open source king 👑

**Comment by chris-araman** [comment]
I think I've convinced myself that the propagation of provisioning profiles is generally useful.

Unfortunately, I think the App Store Connect API functionality is only useful for self-hosted runners, where private keys persist. Would you be comfortable accepting this PR in order to support the functionality only for self-hosted runners?

Support for GitHub-hosted runners could be added in a later PR by persisting any private keys installed using the App Store Connect API. I want to err on the side of incremental progress, and I worry about letting this PR die on the vine.

**Comment by mxcl** [comment]
Possibly the private key could be written to the repo as a secret?

Either way that can happen later. It’s fine that this only works for private runners provided we document it as such.

**Comment by chris-araman** [comment]
Sounds good. I'll add some clarifying documentation to the README and YAML, then merge. Probably tonight or tomorrow. I can (minor) release, too, if you like.

**Pull request merged**